### ISSUE TITLE
razers3: cigar.h:510 Assertion failed

### ISSUE THREAD

**Issue by pipecedeno** [seqan/seqan#2474]
Hello, I'm trying to run an alignment as part of a flow that I'm making, and for some reason while aligning some kmers to a genome some of the alignments can't be performed, as it throws the following error:

```
home/luis/programs/seqan/include/seqan/bam_io/cigar.h:510 Assertion failed : errors == mdErrors was: 1 != 2

stack trace:
  0    [0x55f4db5c275d]  seqan::ClassTest::fail() + 0x12
  1    [0x55f4db70c97f]  void seqan::_alignAndGetCigarString<seqan::String<seqan::CigarElement<char, unsigned int>, seqan::Alloc<void> >, seqan::String<char, seqan::Alloc<void> >, seqan::Segment<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5Q_>, seqan::Alloc<void> > const, seqan::InfixSegment>, seqan::Segment<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5Q_>, seqan::Alloc<void> >, seqan::InfixSegment>, seqan::AlignedReadStoreElement<long, seqan::GapAnchor<long>, void> const, int, seqan::BamAlignFunctorSemiGlobalGotoh>(seqan::String<seqan::CigarElement<char, unsigned int>, seqan::Alloc<void> >&, seqan::String<char, seqan::Alloc<void> >&, seqan::Segment<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5Q_>, seqan::Alloc<void> > const, seqan::InfixSegment> const&, seqan::Segment<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5Q_>, seqan::Alloc<void> >, seqan::InfixSegment> const&, seqan::AlignedReadStoreElement<long, seqan::GapAnchor<long>, void> const&, int&, seqan::BamAlignFunctorSemiGlobalGotoh&) + 0x1c4
  2    [0x55f4db6c2b4a]  void seqan::alignAndGetCigarString<seqan::String<seqan::CigarElement<char, unsigned int>, seqan::Alloc<void> >, seqan::String<char, seqan::Alloc<void> >, seqan::ContigStoreElement<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5Q_>, seqan::Alloc<void> >, seqan::GapAnchor<long>, void>, seqan::Segment<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5Q_>, seqan::Alloc<void> >, seqan::InfixSegment>, seqan::AlignedReadStoreElement<long, seqan::GapAnchor<long>, void> const, int, seqan::BamAlignFunctorSemiGlobalGotoh>(seqan::String<seqan::CigarElement<char, unsigned int>, seqan::Alloc<void> >&, seqan::String<char, seqan::Alloc<void> >&, seqan::ContigStoreElement<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5Q_>, seqan::Alloc<void> >, seqan::GapAnchor<long>, void> const&, seqan::Segment<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5Q_>, seqan::Alloc<void> >, seqan::InfixSegment> const&, seqan::AlignedReadStoreElement<long, seqan::GapAnchor<long>, void> const&, int&, seqan::BamAlignFunctorSemiGlobalGotoh&) + 0xee
  3    [0x55f4db68dcf8]  void seqan::setPrimaryMatch<MyFragStoreConfig, seqan::FragmentStoreConfig<MyFragStoreConfig>, seqan::AlignedReadStoreElement<long, seqan::GapAnchor<long>, void>, seqan::AlignQualityStoreElement<signed char, void>, seqan::BamAlignFunctorSemiGlobalGotoh>(seqan::BamAlignmentRecord&, seqan::FragmentStore<MyFragStoreConfig, seqan::FragmentStoreConfig<MyFragStoreConfig> >&, seqan::AlignedReadStoreElement<long, seqan::GapAnchor<long>, void> const&, seqan::AlignQualityStoreElement<signed char, void> const&, seqan::BamAlignFunctorSemiGlobalGotoh&, bool) + 0x367
  4    [0x55f4db670ed3]  void seqan::writeAlignments<void, MyFragStoreConfig, seqan::FragmentStoreConfig<MyFragStoreConfig>, seqan::BamAlignFunctorSemiGlobalGotoh>(seqan::FormattedFile<seqan::Tag<seqan::Bam_>, seqan::Tag<seqan::Output_>, void>&, seqan::FragmentStore<MyFragStoreConfig, seqan::FragmentStoreConfig<MyFragStoreConfig> >&, seqan::BamAlignFunctorSemiGlobalGotoh&) + 0x200
  5    [0x55f4db63b08d]  int seqan::dumpMatches<MyFragStoreConfig, seqan::FragmentStoreConfig<MyFragStoreConfig>, seqan::String<seqan::String<unsigned short, seqan::Alloc<void> >, seqan::Alloc<void> >, seqan::RazerSSpec<false, false>, seqan::RazerSMode<seqan::RazerSGlobal, seqan::RazerSGapped, seqan::RazerSErrors, seqan::NMatchesNone_> >(seqan::FragmentStore<MyFragStoreConfig, seqan::FragmentStoreConfig<MyFragStoreConfig> >&, seqan::String<seqan::String<unsigned short, seqan::Alloc<void> >, seqan::Alloc<void> >&, seqan::String<char, seqan::Alloc<void> >, seqan::RazerSOptions<seqan::RazerSSpec<false, false> >&, seqan::RazerSMode<seqan::RazerSGlobal, seqan::RazerSGapped, seqan::RazerSErrors, seqan::NMatchesNone_> const&) + 0x21ec
  6    [0x55f4db611ece]  int seqan::dumpMatches<MyFragStoreConfig, seqan::FragmentStoreConfig<MyFragStoreConfig>, seqan::String<seqan::String<unsigned short, seqan::Alloc<void> >, seqan::Alloc<void> >, seqan::RazerSSpec<false, false>, seqan::RazerSGlobal, seqan::RazerSGapped, seqan::NMatchesNone_>(seqan::FragmentStore<MyFragStoreConfig, seqan::FragmentStoreConfig<MyFragStoreConfig> >&, seqan::String<seqan::String<unsigned short, seqan::Alloc<void> >, seqan::Alloc<void> >&, seqan::String<char, seqan::Alloc<void> >, seqan::RazerSOptions<seqan::RazerSSpec<false, false> >&, seqan::RazerSMode<seqan::RazerSGlobal, seqan::RazerSGapped, seqan::Tag<seqan::Nothing_>, seqan::NMatchesNone_>) + 0x69
  7    [0x55f4db6015c7]  int seqan::dumpMatches<MyFragStoreConfig, seqan::FragmentStoreConfig<MyFragStoreConfig>, seqan::String<seqan::String<unsigned short, seqan::Alloc<void> >, seqan::Alloc<void> >, seqan::RazerSSpec<false, false> >(seqan::FragmentStore<MyFragStoreConfig, seqan::FragmentStoreConfig<MyFragStoreConfig> >&, seqan::String<seqan::String<unsigned short, seqan::Alloc<void> >, seqan::Alloc<void> >&, seqan::String<char, seqan::Alloc<void> >, seqan::RazerSOptions<seqan::RazerSSpec<false, false> >&) + 0x108
  8    [0x55f4db5e9f0d]  int mapReads<seqan::RazerSSpec<false, false> >(seqan::StringSet<seqan::String<char, seqan::Alloc<void> >, seqan::Owner<seqan::Tag<seqan::Default_> > >&, seqan::StringSet<seqan::String<char, seqan::Alloc<void> >, seqan::Owner<seqan::Tag<seqan::Default_> > >&, seqan::RazerSOptions<seqan::RazerSSpec<false, false> >&) + 0xad8
  9    [0x55f4db5c08ac]  main + 0x7cd
 10    [0x7fd0af39dfd0]  /lib/x86_64-linux-gnu/libc.so.6(+0x2dfd0)
 11    [0x7fd0af39e07d]  __libc_start_main + 0x7d
 12    [0x55f4db5b18e5]  _start + 0x25

Aborted (core dumped)
```

The command I'm running is 
```
razers3 -ng -mN -i 95 intermediate/mod_genome/clado_s_genome_2_mod.fasta intermediate/dump_reference/20_reference_genome.fasta -o clado_s_genome_2.sam
```
This command works fine in my computer, but to test my flow I'm running this commands in another computer with ubuntu 20.10 and for some reason this error appears, while in my other computer no problem is prompting.

The version of razers I'm using is the following, which was installed nov 17 (output of razers3 --version):
razers3 version: 3.5.8 [f5f6583]
SeqAn version: 2.4.0

I'm attaching the files I used for this test, just changed the fasta extension because github doesn't recognize fasta formats.

Hope you can help me,
Thanks.

[20_reference_genome.txt](https://github.com/seqan/seqan/files/7559558/20_reference_genome.txt)
[clado_s_genome_2_mod.txt](https://github.com/seqan/seqan/files/7559560/clado_s_genome_2_mod.txt)

**Comment by eseiler** [comment]
Dear @pipecedeno,

thanks for reporting and adding some test data.
I'll investigate and try to reproduce the bug.

It might be that this is already resolved on the `develop` branch; but I didn't immediately see any commits that would indicate a fix in `razers3` or the `bam_io` modules.
If it's **easy** for you to build razers3 from source on the `develop` branch, this would be something to try.

**Comment by pipecedeno** [comment]
Apparently I solved the issue, so it must have been that I made an error while installing razers3, because I changed my Ubuntu version on my machine from 21.10 to 20.04, and reinstalled the software again, and in this case I used "cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_FLAGS:STRING="-march=native"" instead of "cmake .. -DCMAKE_BUILD_TYPE=Release" and didn't copied the bin file to the /usr/local/bin directory, so just added the program to path. And I after doing all of these the error no longer appears. So that's why I think it was an error I made during the installation.

Thanks for your help
